plugin/src/Reconciler/hydrate.lua

```diff
@@ -0,0 +1,50 @@
+--[[
+	Defines the process of "hydration" -- matching up a virtual DOM with
+	concrete instances and assigning them IDs.
+]]
+
+local invariant = require(script.Parent.Parent.invariant)
+
+local function hydrate(instanceMap, virtualInstances, rootId, rootInstance)
+	local virtualInstance = virtualInstances[rootId]
+
+	if virtualInstance == nil then
+		invariant("Cannot hydrate an instance not present in virtualInstances\nID: {}", rootId)
+	end
+
+	instanceMap:insert(rootId, rootInstance)
+
+	local existingChildren = rootInstance:GetChildren()
+
+	-- For each existing child, we'll track whether it's been paired with an
+	-- instance that the Rojo server knows about.
+	local isExistingChildVisited = {}
+	for i = 1, #existingChildren do
+		isExistingChildVisited[i] = false
+	end
+
+	for _, childId in ipairs(virtualInstance.Children) do
+		local apiChild = virtualInstances[childId]
```

Comment:
Why is this called apiChild as opposed to something like virtualChild?

Reply:
At one point, these objects were called ApiInstances instead of VirtualInstances. Looks like something I forgot to update :D